%Preamble
\documentclass[12pt,oneside,letterpaper]{article}

% graphicx package, useful for including eps and pdf graphics
\usepackage{graphicx}
\DeclareGraphicsExtensions{.pdf,.png,.jpg}

% basic packages
\usepackage{color}
\usepackage{parskip}
\usepackage{float}

% make links a more pleasing color and don't box them
\definecolor{blue}{rgb}{0.85,0.70,0.99}
\usepackage[hidelinks]{hyperref}
\hypersetup{colorlinks=true,linkcolor=black,citecolor=black,urlcolor=blue}

% text layout
\usepackage{geometry}
\geometry{textwidth=15cm} % 15.25cm for single-space, 16.25cm for double-space
\geometry{textheight=22cm} % 22cm for single-space, 22.5cm for double-space

% helps to keep figures from being orphaned on a page by themselves
\renewcommand{\topfraction}{0.85}
\renewcommand{\textfraction}{0.1}

% bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,font=small]{caption}

% review layout with double-spacing
%\usepackage{setspace}
%\doublespacing
%\captionsetup{labelfont=bf,labelsep=period,font=doublespacing}

% cite package, to clean up citations in the main text
\usepackage{cite}

% Remove brackets from numbering in list of References
\renewcommand\refname{\large References}
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\usepackage{authblk}
\renewcommand\Authands{ \& }
\renewcommand\Authfont{\normalsize \bf}
\renewcommand\Affilfont{\small \normalfont}
\makeatletter
\renewcommand\AB@affilsepx{, \protect\Affilfont}
\makeatother

% notation
\usepackage{amsmath}
\usepackage{amssymb}

\renewcommand{\vec}[1]{\boldsymbol{#1}}
\newcommand{\Prob}{\mathbb{P}}
\newcommand{\wt}{\text{wt}}
\newcommand{\var}{\text{var}}
\newcommand{\varE}{\text{escape}}
\newcommand{\varT}{\text{transmiss}}
\newcommand{\vac}{\text{vac}}

% Inline comments
\usepackage[normalem]{ulem}
\definecolor{purple}{rgb}{0.459,0.109,0.538}
\def\tbc#1{\textcolor{purple}{[#1]}}

\title{Transmission mechanisms determine relative fitness and frequency dynamics of viral variants}

\author[1,2,*]{Marlin D.\ Figgins}
\author[1,3]{Trevor Bedford}
\affil[1]{Vaccine and Infectious Disease Division, Fred Hutchinson Cancer Center, Seattle, WA, USA}
\affil[2]{Department of Applied Mathematics, University of Washington, Seattle, WA, USA}
\affil[3]{Howard Hughes Medical Institute, Seattle, WA, USA}
\affil[*]{Corresponding author: mfiggins@uw.edu}
\date{\today}
% https://maehler.se/blog/2013/11/01/authors-and-affiliations-in-latex

\begin{document}

\maketitle

\begin{abstract}
    Over the course of the COVID-19 pandemic, SARS-CoV-2 variants have repeated emerged, leading to large variant waves in populations of mixed immune backgrounds.
    Classifying these variant viruses into genetically similar clades and lineages has allowed scientists to track the rise and fall of variants by modeling their frequencies over time.
    These models of frequency dynamics enable inference of variant fitness advantages, however, these models are typically divorced from a direct interpretation at the level of transmission.
    In this paper, we derive existing frequency dynamic models from exponentially growing populations and extend them to show that relative fitness of variants can be interpreted using compartment models of infectious diseases.
    We use these models to highlight several complications when analyzing the frequency dynamics of infectious disease variants, namely the difficulty in discovering fitness mechanisms from frequency data alone.
    \tbc{Reading just the abstract I can't interpret what you mean by `fitness mechanisms'. I'd spell out that you're concerned with inherent transmission advantage vs immune escape.}
    We then develop several models for inference of frequency dynamics using Gaussian Processes and dimensionality reduction techniques to estimate latent pseudo-immune components which explain variant dynamics.
    Finally, we apply these models to publicly available SARS-CoV-2 sequences in the United States \tbc{finding...}.
     % To better understand the relationship between population dynamics expected from traditional compartment models of epidemics,
\end{abstract}

%TODO: References
% Huddleston et al, Antigenic waves, Cecile's paper, Usher, Obermyer, the one Jesse sent (?), Recent one ought of Sarah's lab, another talking about backboosting and original antigenic sin, immune dynamics cell paper
% Tegally, Tulio's lab, BA.2.12.1 paper
% Champredon paper on SE^nI^mR models so that we can extend to models with non-exponential generation times including latent periods .etc


\section*{Introduction}

%TODO: Flesh out, focusing on the emergence of looking at empirical variant advantage with frequency based model due to increased sequencing globally + tools like Nextclade.
%TODO: Include notes of desire to relate these frequency / selection estimates to transmission processes to understand risk posed by emerging variants

%TODO: Intro sentence about SARS-CoV-2

Over the course of the COVID-19 pandemic, there have been repeated large waves of infections globally  \cite{tegally2021detection, Volz2021}.
\tbc{Changed in the abstract as well. Internet tells me `several' means 3--5. At this point I think there have been at least 11 major variant waves and many more minor waves.}
These large epidemic \tbc{Most waves do not `sweep' in a pop gen sense} waves have been typically driven by emerging variants of SARS-CoV-2.
\tbc{Include two sentences here. First sentence says that spread of initial variant viruses (Alpha, Beta, Gamma, Delta) was largely driven by increased intrinsic transmissibility. Second sentence that says that subsequently, post-Omicron variant viruses such as XBB, EG.5.1, JN.1, etc... have largely spread due to immune escape. There won't be a good general citation but can cite 2-3 papers for each sentence discussing specific variants. This early on establishes the key distinction between intrinsic transmissibility and immune escape.}
The phenomenon of rapid viral population turnover is consistent with antigenic evolution and is observed in several other viruses such as seasonal influenza, although SARS-CoV-2 is currently remains an outlier in terms of pace of evolution \cite{Koel2013, Bedford2014, kistler2023atlas}.

Given the rapid emergence and spread of SARS-CoV-2 variants \tbc{again removing `selective sweeps'}, it is important to assess the rate of turnover as this can be an important indicator of variant advantage and in addition to understanding how quickly a particular variant is spreading, it may signal the mechanism by which it spreads.
With the increased temporal and geographical scale of sequencing that has occurred, we have gained more data than available for other circulating viruses giving a unique opportunity for insight into the evolution of SARS-CoV-2.

%TODO: Flesh out the increase in genetic sequencing and wanting to assess variant fitness crudely and quickly
\sout{This has motivated the development of statistical methods that can take advantage of this large pool of genetic data including large scale approximate phylogenetic algorithms, identical sequence based methods, and variants frequency dynamic methods.}
\tbc{I think this preceeding sentance can just be dropped.}

To estimate and understand variant turnover, frequency dynamic models of variant relative fitness, such as multinomial logistic regression \tbc{Helpful to provide concrete examples rather than just broad concept}, have been developed.
These models estimate the relative fitness (or selective advantage) of circulating variant viruses from their frequency in sequencing data, typically represented by counts of variant sequences over time within a geographic region.
Relative fitness in these models is often assumed to be a constant quantity for simplicity and intrinsic to the variant of interest, however this may not be reasonable due to the nature of the transmission process.
Additionally, translating these relative fitness estimates to a population-level transmission advantage (in terms of variant infections per wildtype infection) typically requires assumptions on the generation time of the transmission though this can also be approached by jointly modeling the transmission process alongside frequencies \cite{Wallinga2006, figgins2022sars}.

It has been shown that variant transmission advantages differ geographically and temporally which suggests that variant transmission advantages are not necessarily fixed or determined solely due to biology \tbc{This is a strong statement and requires a citation}.
Existing models which allow for variation in variant transmission advantages generally do not have a mechanistic underpinning for why transmission advantages exist and vary geographically and temporally \cite{figgins2022sars, susswein2023leveraging}.
These non-mechanistic transmission advantages are useful for real-time situation analysis though more work is required to develop theory for how various mechanisms determine viral fitness.
\tbc{Similar statement to the Abstract. There's a lot of talk here about `different transmission mechanisms', but then it boils down to intrinsic transmissibility vs immune escape. I think clearer to be upfront that this is the important distinction. Concreteness aids clarity.}

We introduce a theory of relative fitness and transmission advantages for exponentially growing populations.
We then apply this to several ordinary differential equation models of epidemics to assess how different transmission mechanisms may contribute to variant turnover and describe relative fitness.
Using these models, we then highlight the importance of historical population exposure, immune escape, and intrinsic transmissibility in determining variant turnover as well as short term forecasts of frequency growth.
%TODO: Throw in generation time change? Then I can discuss work of SWP and mention paper where they showed shorter generation times for Omicron (?)
With the knowledge gained from this analysis, we develop several new models of variant frequency dynamics which we use to characterize trade-offs between intrinsic transmissibility and immune escape \tbc{I always think of the ordering here as intrinsic transmissibility then immune escape just based on what we empirically observed for SARS-CoV-2 and generally expect for emerging pathogens --- first host adaptation, then antigenic pressure} for pathogens.
Motivated by the results of the theoretical framework, we develop two new non-parametric models for relative fitness estimation and frequency forecasting, one based on Gaussian processes and the other on latent factor models, and apply these models to publicly available SARS-CoV-2 sequence data.

\section*{Results}

\subsection*{Exponentially growing populations to frequency dynamics}%

We consider a viral population consisting of $V$ exponentially-growing variant viruses each with prevalence $I_{v}$.
Defining the time-varying growth rate for the prevalence of variant $v$ as $r_{v}(t)$, we can represent the prevalence using the following ordinary differential equation:

\begin{equation} \label{eq:inhomo_exp_growth}
    \frac{d I_{v}}{d t} = r_{v}(t) I_{v}(t), \quad v = 1,2, \ldots, V.
\end{equation}

The above differential equation has a known solution in terms of the integral of the time-varying growth rate and initial prevalence as follows:

\begin{align*}
I_{v}(t) = I_{v}(0) \exp\left( \int_{0}^{t} r_{v}(s) ds\right),
\end{align*}
where $I_{v}(0)$ is the initial prevalence of variant $v$.

Now turning to the frequency dynamics of the population, we write the frequency of variant $v$ in the population as  $f_{v}(t) = I_{v}(t) / \sum_{u=1}^{V} I_{u}(t)$.
This allows us to derive an ODE for variant frequency in terms of the variant growth rates using the quotient rule for differentiation:
\begin{align*}
    \frac{d f_{v}}{d t} &= f_{v} \left( \sum_{u=1}^{V} [r_{v}(t) - r_{u}(t)] f_{u} \right)\\
                        &= f_{v} \left( r_{v}(t) - \sum_{u=1}^{V} r_{u}(t) f_{u} \right).
\end{align*}

This system of differential equations resembles a logistic growth equation and can be shown to have the following solution in terms of the initial frequencies $f_{v}(0)$ and the variant growth rates:
\begin{align}
    f_{v}(t) &= \frac{ f_{v}(0) \exp( \int_{0}^{t} r_{v}(s) ds)}{\sum_{u=1}^{V}  f_{u}(0) \exp( \int_{0}^{t} r_{u}(s) ds)}.
\end{align}

This form for the frequency of a given variant will serve as a centerpiece for much of the arguments to follow.
We see that by tracking the rate at which variant viruses are spreading, we can construct the corresponding frequency dynamics without knowing the absolute prevalence of any variant.

\paragraph{Relative frequency and relative fitness}%

Using the above equation for the variant frequencies, we can write the relative frequency of variant $v$ over $u$ as $x_{v,u}(t) = f_{v}(t) / f_{u}(t)$ to see:
\begin{align*}
    x_{v, u}(t) = \frac{f_{v}(t)}{f_{u}(t)} &= \frac{f_{v}(0)}{f_{u}(0)} \exp \left( \int_{0}^{t} [r_{v}(s) - r_{u}(s)] ds \right)\\
                                            &=x_{v,u}(0)\exp \left( \int_{0}^{t} \lambda_{v,u}(s) ds \right).
\end{align*}

Notice this relative frequency change depends on the initial relative frequencies and the \emph{relative fitness} $\lambda_{v,u}(t) = r_{v}(t) - r_{u}(t)$ of $v$ over $u$.
We can notice that this relative fitness is exactly
\begin{align}
\lambda_{v, u}(t) = r_{v}(t) - r_{u}(t) = \frac{d }{d t} \left[\log \left( x_{v,u}(t) \right) \right] = - \lambda_{u,v}(t)
\end{align}

We can see that there is a symmetry is the relative fitnesses and that the associated frequency dynamics depend on the differences between relative fitnesses.
This suggests that absolute fitness (in terms of the growth of infections) may not be inferrable from frequencies alone.
This definition of relative fitness becomes essential in describing various existing modeling approaches for frequency dynamic data and motivates possible extensions since we can represent this models as having the form:
\begin{align}
    f_{v}(t) &= \frac{ f_{v}(0) \exp( \int_{0}^{t} \lambda_{v, \text{pivot}}(s) ds)}{\sum_{u=1}^{V}  f_{u}(0) \exp( \int_{0}^{t} \lambda_{u, \text{pivot}}(s) ds)}
\end{align}
for the relative of $v$ relative to some pivot variant $\text{pivot}$.

%TODO: Enumerate assumptions of this?
\paragraph{Cumulative relative-fitness and frequency change}

Above we saw that within our framework frequency change over time intervals depends only on the cumulative relative fitness over time intervals.
We can then characterize approaches for modeling frequency change then in terms of how they represent, estimate, and forecast these relative fitnesses.
This framework includes various existing methods for analyzing frequency data such as the forecast model of Huddleston et al, mutational fitness model of Obermeyer et al, and multinomial logistic regression for frequency estimation. \cite{Huddleston2020, Obermeyer2022, Annavajhala2021}
%Question: How would I nest Piantham in this framework? Can I?


Though this framework can be used to describe existing statistical methods for frequency modeling, it is also applicable to traditional compartmental models of epidemics.
In fact, applying this ideas to compartmental models enables to see how mechanistic assumptions on the transmission process determine relative fitness of variant viruses.

\subsection*{Transmission mechanisms determine relative fitness of variants}

%TODO: Add a sentence or two about multi strain models of epidemics
Multi-strain models of epidemics have been developed to understand the competition between difference strains of a virus for hosts to infect.
These models have typically been used to explain selective sweeps or seasonality in viruses such as influenza or seasonal coronaviruses \cite{Gog2002, Wen2022}

In order to better understand how frequency dynamics of infectious diseases with multiple co-circulating variants, we can apply the above framework to compartmental models of epidemics which can be written in the form of Equation~\ref{eq:inhomo_exp_growth}.
These models will provide us with an intuition of how selection on the level of strain occurs depending on the assumed transmission mechanism of the underlying epidemic model.

\paragraph{Two-strain SIR}%

For simplicity, we will begin by analyzing a two-strain SIR based model in which the variant viruses can differ by increased intrinsic transmissibility (via $\eta_{T}$) and immune escape against wild-type immunity (via $\eta_{E}$).
This system of 4 ordinary differential equations is written in full below.

%TODO: Add 3.x
\begin{align*}
    \frac{d S}{d t} &= - \beta S I_{\wt} - \beta \eta_{T} S I_{v}\\
    \frac{d I_{\wt}}{dt} &= \beta S I_{v} - \gamma I_{w}\\
    \frac{d I_{v}}{dt} &= \beta \eta_{T} S I_{v} + \beta \eta_{T} \eta_{E} \phi_{\wt} I_{v} - \gamma I_{v}\\
    \frac{d \phi_{\wt}}{dt} &= - \beta \eta_{T} \eta_{E} \phi_{\wt} I_{v}
\end{align*}

Writing that $r_{\wt}(t) = \beta S - \gamma$ and $r_{v}(t) = \eta_{T} \beta  S + \beta \eta_{T} \eta_{E} \phi_{\wt} - \gamma$, we can then write the relative fitnesses as:
\begin{align*}
\lambda_{v,\wt}(t) = (\eta_{T} - 1)\beta S(t) + \eta_{T} \eta_{E} \beta \phi_{\wt}(t).
\end{align*}

From this representation of relative fitness, we can see that given fixed increases to overall transmission ($\eta_{T} > 1$) or immune escape ($\eta_{E} > 0$), the observed fitness boost at the level of variant relative fitness still depends on the proportion of the population at risk for infection.

This model can also be extended to an $n$-strain SIR model where each variant strain $i$ with $2\leq i \leq n$ is described by its own advantage parameters $\theta_{i} = (\eta_{T}^{(i)},\eta_{E}^{(i)})$  relative to the wildtype ($\theta_{1} = (0, 0)$).
In Figure \ref{fig:vis_mechanisms}, we implement and simulate a 3-strain model with wildtype as above, an escape variant $\theta_{2} = (0, \eta_{E})$, and a transmissibility increase variant $\theta_{3} = (\eta_{T}, 0)$.
Further extensions also include allowing for differences in generation time through an additional parameter $\eta_{G}$ which multiples the recovery rate $\gamma$, however it is important to note that this still assumes the generation time is an exponential distribution.


\paragraph{Models of immune escape against heterogeneous backgrounds}%

We'll now consider a model where all hosts are assumed to fall into one of $B$ immune background $\phi_{b}$ for $b =1, \ldots, B$.
We assume that infection by each variant $v$ then leaves recovered hosts into the corresponding immune background $b_{v}$.
Variant transmission then occurs via immune escape against a background leading to a matrix of escape rates $\vec{\eta} = \eta_{v,b}$ for variants $v$ and background $b$.

We can then write the system of ODEs as
\begin{align*}
    \frac{d I_{v}}{dt} &= \beta \sum_{1\leq b \leq B} \eta_{v, b} \phi_{b} I_{v} - \gamma I_{v}, \quad v = 1, \ldots, V\\
    \frac{d \phi_{b}}{dt} &= - \beta \sum_{1\leq v \leq V} \eta_{v,b}\phi_{b} I_{v} +  \sum_{v:\ b_{v} = b} \gamma I_{v}
\end{align*}

With this formulation, we can then write the relative fitnesses in terms of the escape rates $\eta_{v,b}$ and the immune background proportions $\phi_{b}$ :

\begin{equation} \label{eq:escape_relative_fitness}
    \lambda_{v, u}(t) = \beta \sum_{1\leq b \leq B}(\eta_{v,b} - \eta_{u,b}) \phi_{b}(t).
\end{equation}

Using this model of immune escape, we cna see relative fitnesses among variants can be decomposed into differences in immune escape among immune backgrounds within a population.
Due to the dependence here on the proportion of each immune background in determining fitness, this suggests that the overall distribution of susceptibility to strains is potentially an important consideration when translating individual-level measures of immune escape to population-level estimates of variant fitness.
Understanding the size and complexity of this immune space may therefore be useful for parameterization and forecasting of variant frequencies.
However, the extent to which modeling this complexity affects estimates of relative fitness also depends on how quickly the distribution of immune backgrounds change.

Though the derivation above uses a simplified model of using most recent infection to sort individuals into an immune group, we show that a more complicated model that accounts for the entire exposure history of the host also gives a similar decomposition to relative fitness in \ref{ssec:full_immune_history}.


%TODO: Compare this to the model where you explicitly count histories
% These give the same but you are summing over immune histories!

%TODO: Suggest that we might use PCA, SVD, and latent factor models to gain a better understanding of these models

For simplicity, we'll begin with a two variant system with vaccination, so there are three immune backgrounds $\phi_{\wt}, \phi_{\var},$ and $\phi_{\vac}$.

% (Does this `increase' antigenic evolution? What does that even mean?)

\begin{figure}[h]
    \centering
    \includegraphics[width=0.8\linewidth]{./figures/vis_mechanisms.png}
    \caption{\textbf{Method overview.}
    Mechanistic transmission models constrain variant frequency dynamics by specifying a functional form for relative fitnesses.
    Simulations of a three-variant model including a wildtype, one intrinsic transmission increased variant, and one immune escape variant show the relationship between population-level transmission and selection for variants.
    A. Prevalence by variant.
    B. Exponential growth rate by variant.
    C. Variant frequency.
    D. Relative fitness relative to wildtype.
    E. Underlying immune pools.
    F. Effective reproduction number by variant.
}%
    \label{fig:vis_mechanisms}
\end{figure}

\section*{Applications}

Using the theory developed for exponentially-growing variant populations, we now re-visit existing methods for modeling viral frequency dynamics.

\paragraph{Multinomial Logistic Regression}%

We begin with multinomial logistic regression with fixed relative fitnesses (MLR).
This model can be written as

\begin{align*}
    f_{v}(t) = \frac{f_{v}(0) \exp(\lambda_{v} t)}{\sum_{u} f_{u}(0) \exp(\lambda_{u} t)},
\end{align*}

where $f_{v}(t)$ is the frequency of variant $v$ at time $t$ and $\lambda_{v}$ is the relative fitness of variant $v$.
This provides estimates of the relative fitness compared to some reference strain $u^{*}$for which $\lambda_{u^*} = 0$.
Converting this estimate to an estimate of transmission advantage (relative effective reproduction number) requires assuming a delta distribution of the generation time. \cite{Wallinga2006}

We can then see this model results from assuming that the at-risk populations are constant over-time.
This assumption is useful since it requires no outside knowledge of the at-risk population and relative infection rates, though this may be less useful for longer forecasts or when there is large turnover in at-risk populations due to infection.

\paragraph{Flu-forecasting}%

Motivated by the observed antigenic evolution of seasonal influenza, the authors approximate the cumulative relative fitness between flu seasons on the level of individual strains as:

\begin{align*}
    \Lambda_{v,u}(t + \Delta t,t) = (\beta_{1} x_{v,1} + \cdots + \beta_{p} x_{v, p})\Delta t = (\vec{\beta} \cdot \vec{x}_{v} \Delta t),
\end{align*}

where the relative fitness is determined by strain-specific predictors $\vec{x}_{v}$ and the regression parameter $\vec{\beta}_{v}$ are estimated.
% The authors choose predictors which describe the antigenic properties of strains

\paragraph{Estimating relative fitness using Gaussian Processes}%

We develop a method for using Gaussian processes to model variant relative fitnesses.

%TODO: Provide short explainer on Gaussian processes and use figure from lab meeting as supplemental figure

Here, the relative fitnesses are modeled using an approximate Gaussian process with a Matern 5/2 kernel and shared hyper-parameters across variants.
For efficiency, we implement a Hilbert Space Gaussian Process (HSGP) approximation instead of fitting $V$ independent Gaussian processes since the HSGP allow us to share basis functions \cite{riutortmayol2022practical}.
This model is used in Figure \ref{fig:gp_example} to estimate the relative fitnesses of the simulated variant dynamics shown in Figure \ref{fig:vis_mechanisms}.

\begin{figure}[h]
    \centering
    \includegraphics[width=0.8\linewidth]{./figures/gp_example.png}
    \caption{\textbf{Estimating relative fitness with Gaussian Processes.}
    Gaussian processes allow us a non-parametric estimate of the relative fitness for variants.
    Below we show an example of using Gaussian processes to model the 3 variant example shown in Figure \ref{fig:vis_mechanisms}.
    A. Synthetic sequence counts generated using a multinomial distribution with frequencies from Figure \ref{fig.vis_mechanisms}C.
    B. Frequencies and posterior frequencies according to Gaussian process model. Intervals show the 80\% credible interval.
    C. Relative fitnesses. Dashed line shows true relative fitnesses from mechanistic model.
}
    \label{fig:gp_example}
\end{figure}


\paragraph{Transmisibility-Escape Tradeoff}%

Extending this analysis to two variant viruses $v= \wt, \varE, \varT$ such that $\eta_{T}^{\varE} = 1$ and $\eta_{E}^{\varT} = 0$, we can write relative fitnesses an escape only variant or transmissibility increased variant as
\begin{align*}
    \lambda_{\varE, \wt} &= \eta_{E} \beta \phi_{\wt}(t)\\
    \lambda_{\varT, \wt} &= (\eta_{T} - 1) \beta S(t).
\end{align*}

In the simplest case where individuals are either susceptible or have wildtype immunity ($S(t) + \phi_{w}(t) = 1$), we can compute the critical immune fraction $\phi^{*}$ at which $\lambda_{\varE, w}(\phi^{*}) = \lambda_{\varT, w}(\phi^{*})$ as
\begin{equation} \label{eq:critical_immunity}
    \phi^{*} = \frac{\eta_{T} - 1}{\eta_{E} + \eta_{T} - 1}.
\end{equation}

For past exposure level greater than $\phi^{*}$, escape variant have a higher relative fitness.
This trade off shows that increasing escape rate entails that a lower proportion of past exposure is needed for escape variants to be preferred (Figure \ref{fig:transmission_tradeoff}).
Additionally, this shows that when intrinsic transmissibility increases are limited escape is more likely to be a dominant mechanism for variant turnover.

We visualize this trade-off in Figure \ref{fig:transmission_tradeoff}

\begin{figure}[h]
    \centering
    \includegraphics[width=0.8\linewidth]{./figures/transmission_tradeoff.png}
    \caption{\textbf{Trade-off between immune escape and transmissibility increases.}
    A. Relative fitness for a transmissibility increasing variant and an immune escaping variant for $\eta_{T}=0.3$, $\eta_{E}=1.2$, $\beta=0.875$.
    The intersection point shows the point after which the escape variant is has higher fitness.
    B. The critical exposure proportion is shown for various escape fraction and transmissibility increase. Above the critical exposure proportion, we expect dominance of escape variants.
    C. The minimum escape fraction needed for second waves to be comprised of escape variant assuming competition with transmissibility increase variants and first wave with a given $R_{0}$.}
    \label{fig:transmission_tradeoff}
\end{figure}

\paragraph{Are initial growth rates sufficient for predicting dominance}%

Main points: though our equations show that things depend on distribution of immune backgrounds, a simple linear model should be able to replicate relative fitness given escape estimates if the distribution of immune backgrounds is unchanged in time.
We use the models developed before to gain some idea on the time-horizon of this assumption in the absence of the emergence of new escape variants.

What happens when there is no mutation? Practical limits given mutation dynamics?
Robustness of relative fitness under similar and varying mechanisms.

We can see this by considering a Taylor expansion of the relative fitness about the point of estimation $t_{0}$:

\begin{align*}
    \lambda_{v,u}(t) \approx \lambda_{v,u}(t_{0}) + \sum_{1\leq b \leq B} (\eta_{v, b} - \eta_{u,b}) \frac{d\phi_b}{dt}(t_0).
\end{align*}

%TODO: Can we use to this to evaluate time horizon on turnover?
%TODO: We could compare assuming constant to real fitness value and see how that affects estimates over time?
%TODO: Can then represent this as a time to x% error in frequency assuming no new emergence. Would be interesting to try this for SARS-2 and flu like parameters

% For example, we can consider the case where the distribution of immune backgrounds is nearly unchanged in time $\frac{d\phi_{b}}{dt} \approx 0$

% \paragraph{How does changing immune pools affect selection?}%
% - This is looking at vaccination as a driver of selection

% Basically, we want to understand $\lambda(\phi_{1}, \ldots, \phi_{S})$. What do we want to say?

% To understand how change in immune-landscape in the host population affects the relative fitness of immune escape variants, we can consider the gradient of an immune escape model

% Using equation \ref{eq:escape_relative_fitness}, we can write the gradient of the relative fitness with respect to the immune backgrounds as

% \begin{align*}
%     \nabla_{\phi} \lambda^{\text{escape}}_{v,u}(t) = (\eta_{v,1} - \eta_{u,1}, \ldots, \eta_{v,B} - \eta_{u,B}).
% \end{align*}

% From this equation, it's clear that to decrease


% QUESTION: Can vaccination push you into the regime where escape is preferred? Yes! When is that better overall tho at the population level?
% FIGURE IDEA: For varying vaccination rates, show when escape or transmission is preferred.
% FIGURE IDEA: Show based on simulation with rates probability escape variant wins.

% This probability exists in closed form?

% However, decreasing relative fitness for a variant relative to another is not the same as decreasing selection over.
% In order to decrease,

% Do we have an idea for variant pressure / total selection here? Can we show whether decreasing selection is same as decreasing cases?

% Do we want to consider an expansion in $\frac{d}{d\phi_{s}}$ to assess sensitivity to immune group? What direction increases section overall? (This will come down to just the eta?)
% You can have higher selection, but how do cases change in simulation. What would vaccinating against wild-type do? What would vaccinating against variant do? Depends on escape potential against variant?
% How does vaccination uptake matter here? Do we want to plot change in relative fitnesses as a function of update? Increased selection but decreased cases

% Basically, we want to also monitor decrease in absolute fitness because increased selection pressure =/= increased cases


% Is this not the same as the transmissibility trade off? Or are we considering two immune escaping?
% We want to think explicitly about vaccination and selection for variants?
% Could we treat this similar to SWP leaky paper where some fraction is reset or leaky vaccine to see what changes depending on vaccine efficacy
% Basically, vaccination can push you into the zone where escape is preferred :O (SEE ABOVE)

\paragraph{Correlations are insufficient for mechanism identification}%

Main points:
Variant growth advantages can be correlated with vaccination proportions even in the absence of immune escape.
Simulated multiple epidemics of a pure increased transmissibility variant at varying levels of initial vaccination and was able to see that estimated growth advantages (using logit-linear model) were correlated with initial vaccination fraction.

We find that you can estimate association of increased relative fitness due to vaccination in the absence of immune escape.
This appears to hold up until both variants' initial effective reproduction number is below 1 i.e. $V_{0} > 1 / R_{v}$.

\begin{figure}[h]
    \centering
    \includegraphics[width=0.8\linewidth]{./figures/correlation_not_mechanism.png}
    \caption{\textbf{Relative fitness is correlated with vaccination levels in the absence of immune escape.}
    We simulate the growth of a pure transmissibility increased variant at varying levels of vaccination.
    Darker colors represent lower vaccine uptake.
    We identify an early growth period where relative fitness is at its highest; the cutoff for this period is denoted with a vertical dashed line.
        A. Prevalence of variant, each line is its own simulation.
        B. Frequency of variant.
        C. Relative fitness for variant over time.
        D. Estimated log growth advantage using linear regression of log relative frequency of variant over wildtype using only data before the early cutoff.
        E. Same as D. but using data from the entire period shown. Notice the drop off of relative fitness above $1 / R_{v}$ at which both variants are declining.
    }
\label{fig:mechanism_identification}
\end{figure}


% \paragraph{Predictive capabilities of simple models on complex dynamics}%

% \paragraph{Dimensionality of immune space}%

% We apply dimensionality reduction techniques to relative fitness estimates generated by using the Gaussian process model to estimate the number of latent dimensions needed to explain the estimated relative fitness dynamics.

%MF: Happy to cut this section, but wanted to look at how many latent factors are needed to explain relative fitnesses estimated by the Gaussian process for certain pandemic periods

%TODO: Show PCA, SVD, or regularized SVD results>

\paragraph{Latent factor models of relative fitness}

The representation of relative fitness using discrete immune backgrounds suggests that there may be low dimensional structure to variant relative fitness.
To generate pseudo-estimates of this latent factors, we develop and implement our method for latent factors models of relative fitness.

We apply this method to data from 37 US states over the period of January 2021 to March 2022. Results focusing on Washington state are shown in Figure~\ref{fig:latent_immune}
%TODO: Extend this section to use more recent data, embed pango lineages?

\begin{figure}[h]
    \centering
    \includegraphics[width=0.8\linewidth]{./figures/latent_immune.png}
    \caption{\textbf{Latent factor models of immunity describe variant structure.}
        We fit the latent immunity factor model to early SARS-CoV-2 frequency data.
        A. Variant frequency. Points are the sample frequencies and the thick lines are modeled variant frequencies. Each is colored by variant of concern.
        B. Estimated relative fitness for each variant.
        C. Estimated pseudo-immunity over time.
        D. Estimated pseudo-escape rates for each variant.
        E. Dimensionality-reduced pseudo-escape rates using Multidimensional Scaling.
    }
\label{fig:latent_immune}
\end{figure}

%TODO: Could be useful to directly compare to published antigenic cartography work as mentioned before ...

\section*{Discussion}

We find that multistrain mechanistic models of epidemics can be used to inform variant turnover at the level of frequency.
This enabled us to illustrate the trade-off between transmissibility increase and immune escape in environments with past exposure.
These mechanistic models also suggest a structure in which relative fitnesses are determined by common causes such as time-varying immunity classes.
Though there appears to be limits to our ability to attribute dynamics to particular mechanisms, the structure of these models suggests that we may improve forecasting by incorporating information about assumed transmission mechanisms.
These ideas are already represented in existing literature through the form of long-term forecast models which incorporate phylogenetic and molecular data which is thought to be indicative of selection and immune escape. (CITE John, LL,  ...)
Future work should seek to identify and incorporate informative data sources for relative fitness estimation and forecasting for particular pathogens alongside the development of the models themselves.

In the absence of data that can inform relative fitness, there is still hope however.
Relative fitnesses can still be estimated and forecasted in the short-term using frequency data alone.
Existing models such as the multinomial logistic regression and the models used here such as the Hilbert Space Gaussian Process and Latent Factor relative fitness models can be used to learn trends present in data to project frequencies to the future, but this may not be reliable for longer terms forecasts due to the emergence of new variants.
These models can still be better extended for short-term forecasts or adapted for longer time horizons.
For example, to account for more complicated patterns in relative fitness, the Hilbert Space Gaussian Process models to non-stationary kernels for scalable inference.
Additionally, though motivated by models of immune escape, latent variable methods may not estimate latent factors that are truly indicative of immune escape directly.

Generally, the structure of the models discussed here are described by simple mechanisms and more work is needed to adapt this framework to complicated epidemic models such as those with non-exponential generation times could be useful.
% Generation time assumptions for compartmental epidemic models
%TODO: Discuss extensions to Laplace transform method
    % We can 1.8 from my general exam to compute roots
    % We can then compute fitness


Futhermore, the models considered here are primarily deterministic in nature and do not explicitly model the emergence of variant viruses only the dynamics after their successful introduction.
In reality, there are biological constraints on the types of variants that are produced in nature and even if there is a "true" fitness boost, the changes for stochastic extinction remains.
That being said, models which account for the appearance of genetic variants through mutation exist, often assuming a latent antigenic space with which fitness is determined. (Cobey, Lassig)

Despite the limitations discussed above, the framework presented here illustrates the importance of transmission mechanism in the evolution of pathogens, highlighting the importance of both variant characteristics and population susceptibility in selection.
We suggest that future models of pathogen transmission and evolution attempt to incorporate current knowledge of the transmission process into their forecasts.

In particular, for antigenically evolving viruses such as influenza and SARS-CoV-2, we suggest that incorporating knowledge on both the diversity in human immune response as well as the escape potential of emerging variants could prove useful in improving long-term forecasts of the viral population.

\section*{Methods}

\paragraph{Generating sequence counts}%

We prepared sequence count data sets using the Nextstrain-curated SARS-CoV-2 sequence metadata \cite{Hadfield2018} which is created using the GISAID EpiCoV database \cite{khare2021gisaid}.
These sequences were tallied according to either their annotated Nextstrain clade or Pango lineage depending on the data set to produce sequence count for each variant and for each day over the period of interest.
%TODO: Flesh out data prep for individual anaylses

\cite{aksamentov2021nextclade}

\paragraph{Likelihood of sequence counts given frequencies}

The models discussed in this paper use counts of variant sequences to inform the underlying variant frequency in the population.
This is accomplished using a multinomial likelihood, so that given counts of sequences $C_{t,v}$ of variant $v$ at time $t$ and total sequences $N_{t}$ collected at time $t$, we have that

\begin{equation*}
    C_{t, \cdot} \sim \text{Multinomial}(N_{t}, f_{v}(t)),
\end{equation*}

where $f_{v}(t)$ is the frequency of variant $v$ at time $t$.
This is a simple model of sequence counts to frequencies and does not account for over-dispersion of sequence counts relative to a multinomial, however, all models can be extended to estimate and account for over-dispersion by replacing the above likelihood with a Dirichlet-Multinomial likelihood.

\paragraph{Approximate Gaussian processes for relative fitness estimation}%

To generate smooth non-parametric estimates of variant growth rates, we develop a Gaussian process based model for relative fitnesses.
That is, we model the relative fitness for each variant $v$ over time as a multivariate normal distribution:

\begin{align*}
    \vec{\lambda}_{v} &\sim \text{Normal}(\vec{\mu}, \vec{\Sigma})\\
    \vec{\Sigma}_{s, t} &= K_{\theta}(s, t),
\end{align*}
where $K_{\theta}$ is a potentially parameterized kernel function.
This induces a structure on the covariance of the relative fitness values over time.

For computational efficiency, we implement a Hilbert Space Gaussian process approximation, so that the relative fitnesses can be represented linearly as

\begin{equation}
    \lambda_{v}(t) \approx \sum_{j=1}^{m} S_{\theta}(\sqrt{\mu_{j}})^{1/2} \cdot \phi_{j}(x) \cdot \beta_{j},
\end{equation}
where $S_{\theta}$ is the spectral density of the kernel $K$, $\mu_{j}$ and $\phi_{j}$ are the eigenvalues and eigenfunctions of the Laplacian, and $\beta_{j} \sim \text{Normal}(0,1)$ \cite{riutortmayol2022practical}.
Since the eigenvalues and eigenfunctions are shared across variants, this allows us to re-use values across variants, simplifying the computation to a matrix multiplication as

%TODO: Fix this
\begin{equation*}
    \vec{\lambda}_{t,v} = \vec{\Phi}\vec{S}_{\theta}\vec{\beta}.
\end{equation*}


\paragraph{Latent immune factor model}%

We show that relative fitnesses dynamics ought to be explained by low-dimensional immunity when transmission dynamics are described with compartment models.
This motivated a model to learn this low-dimensional structure that is inspired by latent-factor models. We start by assuming that the relative fitness of variant $v$ at time $t$ and in geographic location $g$ can be described by $D$ latent factors so that

\begin{equation}
    \lambda_{v}^{g}(t) = \sum_{d=1}^{D} \eta_{v,d} \phi_{d}^{g}(t).
\end{equation}
As the structure here resembles Equation~\ref{eq:escape_relative_fitness}, we call the $\eta$ ``pseudo-escape'' and $\phi_{d}^{g}$ the ``pseudo-immunity''.
To make this more consistent with our intuition here, we model $\phi_{d}^{g}$ to be in $[0,1]$ and model it as smoothly varying in time.
In the results shown above, we model $\text{logit}(\phi_{d}^{g})$ using 4th order splines with [INSERT KNOTS HERE] knots placed uniformly over the time period modeled.
Though we choose to model tesee latent factors with splines, other models would work here. For example, a good alternative would be the approximate Gaussian processes described above.
Additionally, in order to ensure identifiability of the parameter estimates, we fix some base variant $v^*$ which fitness is defined relative to, so that $\eta_{v^*, d} = 0$ for all $1\leq d\leq D$.
For the same reason, we order the components in the first geography.

\subsection*{Data and code accessibility}

Source code used to generate figures as well as model implementations are available at https://github.com/blab/relative-fitness-mechanisms.

\subsection*{Competing interests}%

\subsection*{Author contributions}
MF conceived the study.
% TB gathered sequence and case count data.
MF designed and implemented inference models.
MF performed the analysis.
% MF, TB interpreted the results.
% MF, TB wrote the paper.


\subsection*{Acknowledgements}%

\bibliographystyle{plos}
\bibliography{relative-fitness-mechanisms}

\newpage

\appendix

\setcounter{figure}{0}
\setcounter{table}{0}
\setcounter{page}{1}
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
\renewcommand{\thepage}{S\arabic{page}}
\renewcommand{\thesubsection}{S\arabic{subsection}}

\section*{Supplemental Appendix}

\subsection{Relative fitness for full immune history models}\label{ssec:full_immune_history}

%TODO: Define powers sets + macro to simplify?
% $\mathcal{P}(M \setminus \{i\})$

We show that the simple background model is consistent with an expanded immune history model.
Beginning with the model from Lazebnik and Bunimovich-Mendrazitsky 2022 \cite{Lazebnik2022}, we consider the differential equation for the individuals with strain infection history $J$ and current infecting strain $i$ $R_{J}I_{i}$
\begin{align*}
\frac{dR_{J} I_{i}}{dt} = - \gamma_{J, i} R_{J} I_{i} + \beta_{J, i} R_{J} \sum_{K \in P(M), i\notin K} R_{K}I_{i}.
\end{align*}

Here, infection can occur from any individual infected with strain $i$ assuming their past immune history does not include $i$ and the infected are any recovered individual with immune history $J$  $R_{J}$.
To compute the strain growth rate, we can sum over all possible immune histories for individuals infected with strain $i$, so that
\begin{align*}
    \frac{d I_{i}}{d t} &= \sum_{J \in P(M), i \notin J} \frac{dR_{J} I_{i}}{dt} \\
                        &= - \gamma_{i} I_{i} + \sum_{J \in P(M), i \notin J} \beta_{i, J} R_{J} \sum_{K \in P(M), i\notin K} R_{K}I_{i}\\
                        &= - \gamma_{i} I_{i} + \sum_{J \in P(M), i \notin J} \beta_{i, J} R_{J} I_{i}\\
                        &= \left(-\gamma_{i} + \sum_{J \in P(M), i \notin J} \beta_{i,J} R_{J} \right) I_{i}\\
                        &= \left(-\gamma + \beta\sum_{J \in P(M), i \notin J} \eta_{i,J} R_{J} \right) I_{i}\\
\end{align*}

Assuming that the transmission rate can be decomposed as a base transmission rate $\beta$ and a strain $i$ and immune history $J$ specific escape rate $\eta_{i, J}$ and that the recovery rate is constant, we notice this is identical to our previous immune background model.
Therefore, our relative fitnesses can be written as

\begin{align*}
    \lambda_{i, j} = \beta \sum_{B \in P(M)} (\eta_{i, B} - \eta_{j, B}) R_{B},
\end{align*}
where for simplicity we define $\eta_{v, B} = 0$ if $v \in B$.


\end{document}
